src/util/ieee_float.cpp

```diff
@@ -123,7 +123,7 @@ std::string ieee_floatt::format(const format_spect &format_spec) const
 mp_integer ieee_floatt::base10_digits(const mp_integer &src)
 {
   mp_integer tmp=src;
-  assert(tmp>=0);
+  PRECONDITION(tmp>=0);
```

Comment:
There _could_ be a subtle distinction. 
In this location this could be considered as getting two birds with one stone - a check on src, and a check that the copy constructor initialised  the mp_integer appropriately. In which case, it could be a CHECK_RETURN.  Or more appropriately a data invariant.
This might however be somewhat confusing. 

As all the operations are on tmp, it seems appropriate to check tmp.